vmm/src/vm.rs

```diff
@@ -1023,7 +1098,14 @@ impl Vm {
                 let cmdline = Self::generate_cmdline(payload)?;
                 Self::load_kernel(kernel, Some(cmdline), memory_manager)
             }
-            _ => Err(Error::InvalidPayload),
+            _ => {
+                #[cfg(feature = "igvm")]
+                if let Some(_igvm_file) = &payload.igvm {
+                    let igvm = File::open(_igvm_file).map_err(Error::IgvmFile)?;
+                    return Self::load_igvm(igvm, memory_manager, cpu_manager);
+                }
+                Err(Error::InvalidPayload)
+            }
```

Comment:
I'd prefer a explicit match case for the `igvm` case and leave the default case unchanged.

Reply:
How do I do this with a feature gate, rest of then are not guarded with feature. Also I can't put a feature gate inside match statement. Any suggestion?

Reply:
I see. The way how `igvm` will be loaded is totally different from the rest of payloads. Let's move the `igvm` code block out of the match statement and put it at the beginning of this function. This is cleaner to me.